### ISSUE TITLE
Indexing error place_node_fuzzy_area when boundary is a node

### ISSUE THREAD

**Issue by mtmail** [osm-search/Nominatim#2823]
When indexing Germany `place_node_fuzzy_area` seems to fail when a boundary is mapped as a node. There's only two OSM places with that attribute combination.

```sql
select osm_type, osm_id, country_code, name from placex where class='boundary' and type='postal_code' and rank_address=11 and osm_type='N';
 osm_type |   osm_id   | country_code |      name
----------+------------+--------------+----------------
 N        |  254507405 | de           | "ref"=>"99880"
 N        | 7601064862 | de           | "ref"=>"88693"
```

https://www.openstreetmap.org/node/254507405
https://www.openstreetmap.org/node/7601064862

```
$ nominatim --version
Nominatim version 4.1.0-0 (d8be8a72)
```

Postgresql logs

```sql
ERROR:  Distance must not be greater than 0
CONTEXT:  PL/pgSQL function place_node_fuzzy_area(geometry,integer) line 15 at RETURN
	PL/pgSQL function add_location(bigint,character varying,integer,integer[],integer,integer,text,geometry,geometry) line 21 at assignment
	PL/pgSQL function placex_update() line 456 at assignment
STATEMENT:   UPDATE placex
	                SET indexed_status = 0, address = v.addr, token_info = v.ti,
	                    name = v.name, linked_place_id = v.linked_place_id
	                FROM (VALUES (666501, hstore(ARRAY['ref'], ARRAY['99880'])::hstore, hstore(ARRAY['postcode'], ARRAY['99880'])::hstore, NULL::int, '{"names": "{1820214,1820215}", "postcode": "99880"}'::jsonb)) as v(id, name, addr, linked_place_id, ti)
	                WHERE place_id = v.id
```

Indexing output
```
Done 3940 in 52 @ 75.728 per second - rank 11 ETA (seconds): 276.72
psycopg2.errors.InternalError_: Distance must not be greater than 0
CONTEXT:  PL/pgSQL function place_node_fuzzy_area(geometry,integer) line 15 at RETURN
PL/pgSQL function add_location(bigint,character varying,integer,integer[],integer,integer,text,geometry,geometry) line 21 at assignment
PL/pgSQL function placex_update() line 456 at assignment
```

**Comment by lonvia** [comment]
This passes happily on the OSM servers and on my test server. Could you please provide PostgreSQL and PostGIS versions and mention if you have any changes to the default config files.

**Comment by mtmail** [comment]
It's a PostGIS bug. I will follow up with the PostGIS project. Closing here.

3.3.1 released two weeks ago returns an error, 3.2.0 doesn't.

```
SELECT ST_AsText(ST_Project('POINT(0 0)'::geography, 100000, radians(45.0)));
ERROR:  Distance must not be greater than 0
```

The SQL is the example from the official documentation https://postgis.net/docs/ST_Project.html
